From stern@rowland.harvard.edu Tue Jan 31 07:03:01 2006
Date: Tue, 31 Jan 2006 10:02:55 -0500 (EST)
From: Alan Stern <stern@rowland.harvard.edu>
To: Greg KH <greg@kroah.com>, Andrew Morton <akpm@osdl.org>
cc: Arnaud Patard <arnaud.patard@rtp-net.org>
Subject: [PATCH] uhci-hcd: fix mistaken usage of list_prepare_entry
Message-ID: <Pine.LNX.4.44L0.0601310958570.5380-100000@iolanthe.rowland.org>

A recent update to the uhci-hcd driver invoked the list_prepare_entry
macro incorrectly.  This patch (as646) corrects it.

Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/uhci-q.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/usb/host/uhci-q.c
+++ gregkh-2.6/drivers/usb/host/uhci-q.c
@@ -259,7 +259,7 @@ static void uhci_fixup_toggles(struct uh
 	/* Fix up the toggle for the URBs in the queue.  Normally this
 	 * loop won't run more than once: When an error or short transfer
 	 * occurs, the queue usually gets emptied. */
-	list_prepare_entry(urbp, &qh->queue, node);
+	urbp = list_prepare_entry(urbp, &qh->queue, node);
 	list_for_each_entry_continue(urbp, &qh->queue, node) {
 
 		/* If the first TD has the right toggle value, we don't
